File: app/src/main/java/studio/mandysa/music/ui/fragment/ControllerFragment.java
package studio.mandysa.music.ui.fragment;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import studio.mandysa.music.service.Media;
import studio.mandysa.music.service.PlayManager;
import studio.mandysa.music.R;
import studio.mandysa.music.ui.base.BasePlayFragment;
import studio.mandysa.music.utils.ImageLoader;

public class ControllerFragment extends BasePlayFragment {

    private TextView mSongName;

    private ImageView mCover;
    private ImageView mPlayOrPause;

    void setTitle(String title) {
        mSongName.setText(title);
    }

    @Override
    public void setPlayModel() {
        mPlayOrPause.setImageResource(R.mipmap.ic_play);
    }

    @Override
    public void setPauseModel() {
        mPlayOrPause.setImageResource(R.mipmap.ic_pause);
    }

    @Override
    protected void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mSongName = findViewById(R.id.tv_controller_songName);
        mCover = findViewById(R.id.iv_controller_cover);
        mPlayOrPause = findViewById(R.id.iv_controller_playOrPause);

        setPlayModel();

        ImageView mNextSong = findViewById(R.id.iv_controller_nextSong);

        mNextSong.setImageResource(R.mipmap.ic_skip_next);

        mPlayOrPause.setOnClickListener(p1 -> {
            if (getIsPlaying()) {
                PlayManager.getInstance(ControllerFragment.this).pause();
            } else {
                PlayManager.getInstance(ControllerFragment.this).play();
            }
        });
        mNextSong.setOnClickListener(p1 -> PlayManager.getInstance(ControllerFragment.this).skipToNext());

    }

    @Override
    protected int onCreateView() {
        return R.layout.controller;
    }

    @Override
    protected void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        PlayManager.getInstance(this).getMediaMetadataLiveData().observe(this, p1 -> {
            if (p1 == null) return;
            setTitle(p1.getString(Media.TITLE));
            //mMusicView.setSubTitle(p1.getString(MediaMetadata.METADATA_KEY_ARTIST));
        });
        getShareViewModel().picUrl.observe(this, p1 -> ImageLoader.getInstance().load(p1).into(mCover));
    }

}

File: app/src/main/java/studio/mandysa/music/ui/controller/ControllerFragment.kt
package studio.mandysa.music.ui.controller

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.nostra13.universalimageloader.core.ImageLoader
import mandysax.fragment.Fragment
import studio.mandysa.music.R
import studio.mandysa.music.databinding.FragmentControllerBinding
import studio.mandysa.music.logic.utils.bindView
import studio.mandysa.music.logic.utils.getInstance

class ControllerFragment : Fragment() {

    private val mBinding: FragmentControllerBinding by bindView()

    private val mImageLoader = ImageLoader.getInstance()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return mBinding.root
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        getInstance().changeMusicLiveData().observe(viewLifecycleOwner) {
            mBinding.apply {
                controllerTitle.text = it.title
                mImageLoader.displayImage(it.coverUrl, controllerCover)
            }
        }
        getInstance().pauseLiveData().observe(viewLifecycleOwner) {
            mBinding.controllerPlayOrPause.setImageResource(if (it) R.drawable.ic_play else R.drawable.ic_pause)
        }
        mBinding.controllerPlayOrPause.setOnClickListener {
            getInstance().apply {
                if (pauseLiveData().value == true)
                    play()
                else pause()
            }
        }
        mBinding.controllerNextSong.setOnClickListener {
            getInstance().skipToNext()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mImageLoader.clearMemoryCache()
    }

}